content/docs/reference-react-dom-client.md

```diff
@@ -38,7 +38,7 @@ React supports all modern browsers, although [some polyfills are required](/docs
 ### `createRoot()` {#createroot}
 
 ```javascript
-createRoot(container[, options]);
+createRoot(container, [ options ]);
```

Comment:
I think this is mostly an issue with where you got introduced first to this syntax for optional parameters. I'm mostly familiar with `[, options]` but I've also seen `, [options]` e.g. https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/named-and-optional-arguments)

I've seen plenty of people struggling with this syntax (also because it looks like array syntax).

MDN does not have special syntax and just states in prose that a parameter is optional (e.g.`eventInitDict` in https://developer.mozilla.org/en-US/docs/Web/API/SubmitEvent/SubmitEvent)

@gaearon Is this syntax for optional parameters still in use in the Beta docs or do you know if you settled on a specific syntax for optional parameters?

Reply:
In beta we just write multiple signatures after newlines, I think.

Reply:
I believe we aren’t currently using the same syntax in beta docs. Also the syntax in the older documentation looks fine. We use  [, and not ,[ to denote optional params just like mdn docs

Reply:
> We use [, and not ,[ to denote optional params just like mdn docs

Do you have an example for that? The articles I linked don't have special syntax e.g. https://developer.mozilla.org/en-US/docs/Web/API/SubmitEvent/SubmitEvent.

The important part here is that the chose syntax should not be ambigious. Not using any special syntax would fit that requirement IMO. Just like using multiple signatures.

Reply:
I've seen the same in multiple places as well. 
like: 
https://developer.mozilla.org/en-US/docs/Web/API/console/log#syntax
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/onUpdated#syntax 

> The important part here is that the chose syntax should not be ambigious. Not using any special syntax would fit that requirement IMO. Just like using multiple signatures.

Agreed

Reply:
Got it and thanks @harish-sethuraman for mentioning #3919 . I got confused with that [, options ] syntax.